cfgmgr/natmgr.cpp

```diff
@@ -0,0 +1,6391 @@
+    return false;
+}
+
+/* To check the given Static NAT entry is matched with any Static NAPT entry */
+bool NatMgr::isMatchesWithStaticNapt(const string &global_ip, string &local_ip)
+{
+    /* Get all Static NAPT entries */
+    for (auto it = m_staticNaptEntry.begin(); it != m_staticNaptEntry.end(); it++)
+    {
+        vector<string> keys = tokenize((*it).first, config_db_key_delimiter);
+
+        /* Ensure that interface is having some port, otherwise Entry is not yet added */
+        if ((keys[0] == global_ip) && ((*it).second.local_ip == local_ip) &&
+            ((*it).second.interface != NONE_STRING))
+        {
+            /* Matches with the Static NAPT entry */
+            return true;
+        }
+    }
+    return false;
+}
+
+/* To check the given Static NAPT entry is matched with any Static NAT entry */
+bool NatMgr::isMatchesWithStaticNat(const string &global_ip, string &local_ip)
+{
+    /* Get all Static NAT entries */
+    for (auto it = m_staticNatEntry.begin(); it != m_staticNatEntry.end(); it++)
+    {
+        /* Ensure that interface is having some port, otherwise Entry is not yet added */
+        if (((*it).first == global_ip) && ((*it).second.local_ip == local_ip) &&
+            ((*it).second.interface != NONE_STRING))
+        {
+            /* Matches with the Static NAT entry */
+            return true;
+        }
+    }
+    return false;
+}
+
+/* To get the configured interface for the given global ip address */
+bool NatMgr::getIpEnabledIntf(const string &global_ip, string &interface)
+{
+    /* Get all Ports from Ip Interface Info */
+    for (auto it = m_natIpInterfaceInfo.begin(); it != m_natIpInterfaceInfo.end(); it++)
+    {
+        /* Get all IpAddress from Ip Interface Values from the key (Port) */
+        for (auto ipPrefix = m_natIpInterfaceInfo[(*it).first].begin(); ipPrefix != m_natIpInterfaceInfo[(*it).first].end(); ipPrefix++)
+        {
+            /* Check the global ip address is in subnet */
+            if (isGlobalIpMatching(*ipPrefix, global_ip) == true)
+            {
+                /* Matched with this interface, return now */
+                interface = (*it).first;
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+/* This is ideally called on docker stop  */
+void NatMgr::cleanupPoolIpTable(void)
+{
+    SWSS_LOG_INFO("Cleaning the NAPT Pool IP table from APP_DB");
+    for (auto it = m_natPoolInfo.begin(); it != m_natPoolInfo.end(); it++)
+    {
+        /* Delete pool ip from APPL_DB */
+        setNaptPoolIpTable(DELETE, ((*it).second).ip_range, ((*it).second).port_range);
+    }
+}
+
+/* To Add/Delete NAPT pool ip table to APPL_DB */
+void NatMgr::setNaptPoolIpTable(const string &opCmd, const string &ip_range, const string &port_range)
+{
+    uint32_t ipv4_addr_low, ipv4_addr_high, ip, setIp;
+    char ipAddr[INET_ADDRSTRLEN];
+    std::vector<swss::FieldValueTuple> values;
+
+    if (!port_range.empty() and (port_range != "NULL"))
```

Comment:
First check if not NULL?

Reply:
No, When pool is created with only Ip address, "nat_port" value is set to "NULL" string like below

root@sonic:/home/admin# config nat add pool p1 65.55.42.5
root@sonic:/home/admin# redis-cli -n 4
127.0.0.1:6379[4]> keys *NA*
1) "NAT_POOL|p1"
127.0.0.1:6379[4]> hgetall NAT_POOL|p1
1) "nat_port"
2) "NULL"
3) "nat_ip"
4) "65.55.42.5"
127.0.0.1:6379[4]> exit
root@sonic:/home/admin#